src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let clone = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(clone, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete clone[key];
+        }
+        break;
+
+      case 'clear':
+        clone = {};
+        break;
+
+      default:
+        throw new Error('your error text');
```

Comment:
```suggestion
        throw new Error('your error text');
```

It's better to use smth like 'Unknown action type' instead of 'your error text'